apps/src/Sounds.js

```diff
@@ -390,6 +390,25 @@ Sounds.prototype.unmuteURLs = function() {
   this.isMuted = false;
 };
 
+/**
+ * Stop all currently playing sounds, and keep track of the paused sounds so
+ * they can be restarted later.
+ */
+Sounds.prototype.pauseSounds = function() {
+  this.pausedSounds = Object.keys(this.soundsById).filter(
+    soundUrl => this.soundsById[soundUrl].isPlaying_
+  );
+  this.pausedSounds.forEach(soundUrl => this.stopPlayingURL(soundUrl));
```

Comment:
not something i think we necessarily need to do, but an interesting javascript bit: when function signatures match exactly like this, there's a shorthand we can use here

```javascript
this.pausedSounds.forEach(this.stopPlayingURL);
```

you just have to be careful with `this` (not an issue here), whereas using an arrow function ensures `this` is what we expect it to be in this scope

Reply:
hmm, I think `this` binding does actually matter here.
If we do `this.pausedSounds.forEach(this.stopPlayingURL);`, `this` will be `undefined` in `stopPlayingURL`. This is because " if the value of this is not set when entering an execution context, it remains as undefined" ([doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this))


We'd need to do either `this.pausedSounds.forEach(this.stopPlayingURL.bind(this));` or `this.pausedSounds.forEach(this.stopPlayingURL, this);` to make sure `this` is bound correctly. I think both of those are a little less clear than using the arrow function, so I think I'll leave it as is.

Reply:
ohhh, interesting -- i thought `this` would be defined, but maybe i'm mistaking that because that's how i'd expect a class to work, whereas this is a prototype method.

yeah, either way i think what you have is preferred